articles/quickstart/webapp/rails/interactive.md

```diff
@@ -0,0 +1,169 @@
+
+Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
+:::
+::::
+
+## Add Logout to Your Application {{{ data-action=code data-code="app/controllers/auth0_controller.rb#21:40" }}}
+
+Now that you can log in to your Rails application, you need [a way to log out](https://auth0.com/docs/logout/guides/logout-auth0). Revisit the Auth0 controller you created earlier, and finish up the logout method added there.
+
+:::note
+To test this after the previous login step, you will need to clear out your session and then redirect the user to the Auth0 logout endpoint.
+:::
+
+To clear out all the objects stored within the session, call the `reset_session` method within the `auth0_controller/logout` method. [Learn more about reset_session here](http://api.rubyonrails.org/classes/ActionController/Base.html#M000668).
+
+::: note
+The final destination URL (the `returnTo` value) needs to be in the list of `Allowed Logout URLs` . See the [logout documentation](/logout/guides/redirect-users-after-logout) for more.
+:::
+
+Similar to logging in, the user will now be able to logout of your application by visiting the `/auth/logout` endpoint.
+
+```erb
+  <!-- Place a logout button anywhere on your application -->
+  ${"<%= button_to 'Logout', 'auth/logout', method: :get %>"}
+```
+
+::::checkpoint
+::: checkpoint-default
+Add a button to your application that redirects the user to the `/auth/logout` endpoint when clicked. Verify that you are redirect to Auth0 and then quickly back to your application, and that you are no longer logged in.
+:::
+
+::: checkpoint-failure
+Sorry about that. Here are a couple of things you can check:
+
+- Ensure that the correct URLs have been set in your Auth0 client as per the first step in this quickstart
+- Check that the routes have been set up and the Auth0 configuration has been set up in your app
+- [Check the logs](https://manage.auth0.com/#/logs) for any other errors or messages that may have prevented login from working
+
+Still having issues? Check out our [documentation](https://auth0.com/docs) or visit our [community page](https://community.auth0.com) to get more help.
+:::
+::::
+
+## Show User Profile Information {{{ data-action=code data-code="app/controllers/concerns/secured.rb" }}}
+
+To display the user's profile, your application should provide a protected route. You can use a controller `concern` to control access to routes. Create the following file `./app/controllers/concerns/secured.rb`
+
+After you have created the secured controller concern, include it in any controller that requires a logged in user. You can then access the user from the session `session[:userinfo]`, as in the following example:
+
+```ruby
+class DashboardController < ApplicationController
+  include Secured
+
+  def show
+    # session[:userinfo] was saved earlier on Auth0Controller#callback
+    @user = session[:userinfo]
+  end
+end
+```
```

Comment:
I felt the reading experience was better leaving these in (this snippet and the next one) as they're more "examples" than actual files the reader should create, but happy to amend. This is tricky to do nicely when you can only supply external code snippets for H2 headings, as at that point it has an impact on readability.